Tutorials/intro-to-QUAlibrate/calibrations/readme.md

```diff
@@ -0,0 +1,249 @@
+    node.namespace["job"] = job
+
+    keys = [...]
+    data_fetcher = fetching_tool(job, data_list=keys, mode="wait_for_all")
+    values = data_fetcher.fetch_all()
+
+    for key, value in zip(keys, values):
+        node.results[key] = value
+```
+> 📥 This step connects to the OPX, runs the program, fetches the data, and stores results in node.results.
+
+
+### 7️⃣ Data Loading 
+If reusing saved results:
+```python 
+@node.run_action(skip_if=node.parameters.load_data_id is None)
+def load_data(node):
+     node.load_from_id(node.parameters.load_data_id)
+```
+> 📁 Useful for loading previously saved datasets during debugging or reanalysis.
+
+
+### 8️⃣ Data Analysis 
+Wrap your analysis logic inside a dedicated run action. 
+
+You can optionally move your processing and fitting functions to `calibration_utils/time_of_flight/analysis.py` for modularity and reuse — but you're also free to define them directly inside `analyse_data()` function.
+```python
+@node.run_action(skip_if=node.parameters.simulate)
+def analyse_data(node):
+    ...
+```
+> 🧠 This step processes the data.
+
+
+### 9️⃣ Data Plotting
+Wrap your plotting logic inside a dedicated run action. 
+As with the analysis step, you can optionally move your plotting functions into `calibration_utils/time_of_flight/plotting.py` for reuse and cleaner code organization. However, it’s equally valid to define plotting code directly inside `plot_data()` function.
+```python
+@node.run_action(skip_if=node.parameters.simulate)
+def plot_data(node):
+    ...
+```
+> 📊 These figures will show up in Python or in the QUAlibrate Web UI automatically.
+
+
+### 🔟 Save Results
+
+At the end of the node workflow, you should always include a @node.run_action() that calls node.save(). This ensures that the node’s parameters, results, figures, and metadata are properly stored and made available for later use.
+
+```python
+@node.run_action()
+def save_results(node):
+    node.save()
+
+```
+
+> 💾 This saves:
+> 
+> - All parameters and metadata  
+> - Analysis results and plots
+
+---
+
+## 3. 🔍 Understanding a QualibrationNode
+
+For a detailed breakdown of the internal structure of a typical calibration node, please refer to the [Anatomy of a QualibrationNode](./node_anatomy.ipynb) document. It dissects the `time_of_flights.py` node section by section, explaining the purpose of the common components like imports, initialization, run actions (`@node.run_action`), QUA program creation, data handling, analysis, and saving.
+
+---
+
+## 4. 🚀 Extending the Calibration Library
+
+You can easily extend this library by adding your own custom calibration nodes. To ensure compatibility and maintainability, new nodes should follow the same standardized structure and conventions outlined in the "Resonator Spectroscopy Node Explained" document. This includes:
+
+- Using the `# %%` separators for code cells.
+- Defining parameters in a separate `Parameters` class (usually imported).
```

Comment:
I would explicitly specify `from calibration_utils.time_of_flight import Parameters` for clarity